### PR TITLE
Handled different `engine.state.output` types for LRFinder

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Handled tensors and floats from `engine.state.output` for LRFinder, and cover this with test.

Check list:

- [x] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by KickItLikeShika** [comment]
Mypy checks work fine locally, maybe we should restart the GA workflow

File: ignite/handlers/lr_finder.py [hunk 1]
```diff
@@ -137,6 +137,14 @@ def _reset(self, trainer: Engine) -> None:
     def _log_lr_and_loss(self, trainer: Engine, output_transform: Callable, smooth_f: float, diverge_th: float) -> None:
         output = trainer.state.output
         loss = output_transform(output)
+        if not isinstance(loss, float):
+            if isinstance(loss, torch.Tensor):
```
Comment (vfdev-5) [inline ignite/handlers/lr_finder.py:141]: We have to check that tensor is 0-d or 1-d with one-element before applying `.item()`.

Comment (KickItLikeShika) [reply ignite/handlers/lr_finder.py:141]: I missed that, thanks for pointing that out!

File: ignite/handlers/lr_finder.py [hunk 2]
```diff
@@ -137,6 +137,14 @@ def _reset(self, trainer: Engine) -> None:
     def _log_lr_and_loss(self, trainer: Engine, output_transform: Callable, smooth_f: float, diverge_th: float) -> None:
         output = trainer.state.output
         loss = output_transform(output)
+        if not isinstance(loss, float):
+            if isinstance(loss, torch.Tensor) and (loss.ndimension() == 0):
```
Comment (vfdev-5) [inline ignite/handlers/lr_finder.py:141]: You are still missing 1-d with one-element case

File: ignite/handlers/lr_finder.py [hunk 3]
```diff
@@ -137,6 +137,17 @@ def _reset(self, trainer: Engine) -> None:
...
+        if not isinstance(loss, float):
+            if isinstance(loss, torch.Tensor) and (
+                (loss.ndimension() == 0) or (loss.ndimension() == 1 and len(loss) == 1)
+            ):
+                loss = loss.item()
+            else:
+                raise TypeError(
+                    "output of the engine should be of type float or 0d torch.Tensor "
+                    "or 1d torch.Tensor with 1 element, "
+                    f"but got output of type {type(loss).__name__}"
```
Comment (vfdev-5) [inline ignite/handlers/lr_finder.py:149]: This will show "Tensor" but I think we already know that loss is tensor, the point is to show shape maybe here.

File: ignite/handlers/lr_finder.py [hunk 4]
```diff
@@ -137,6 +137,17 @@ def _reset(self, trainer: Engine) -> None:
     def _log_lr_and_loss(self, trainer: Engine, output_transform: Callable, smooth_f: float, diverge_th: float) -> None:
         output = trainer.state.output
         loss = output_transform(output)
+        if not isinstance(loss, float):
+            if isinstance(loss, torch.Tensor) and (
+                (loss.ndimension() == 0) or (loss.ndimension() == 1 and len(loss) == 1)
+            ):
+                loss = loss.item()
+            else:
```
Comment (vfdev-5) [inline ignite/handlers/lr_finder.py:145]: This looks a bit bulky but I have no idea on how to simplify it more than one line.

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the update @KickItLikeShika !

**Review by vfdev-5 — APPROVED** [review]
Thanks @KickItLikeShika !
